src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -392,13 +404,14 @@ private string GenerateRootId()
                 Interlocked.CompareExchange(ref s_uniqPrefix, GenerateInstancePrefix(), null);
             }
 #if DEBUG
-            string ret = s_uniqPrefix + "-" + OperationName + "-" + Interlocked.Increment(ref s_currentRootId).ToString("x") + '.';
+            string ret = s_uniqPrefix + "-" + OperationName.Replace('.', '-') + "-" + Interlocked.Increment(ref s_currentRootId).ToString("x") + '.';
 #else       // To keep things short, we drop the operation name 
             string ret = s_uniqPrefix + "-" + Interlocked.Increment(ref s_currentRootId).ToString("x") + '.';
 #endif
             return ret;
         }
 
+        [SecuritySafeCritical]
```

Comment:
Thanks! it fixed reference issues, but now I have an issue with packaging:

`D:\j\workspace\AllConfigurat---c1564d53\Tools\Packaging.targets(424,5): error MSB4018: The "HarvestPackage" task failed unexpectedly. [D:\j\workspace\AllConfigurat---c1564d53\src\System.Diagnostics.DiagnosticSource\pkg\System.Diagnostics.DiagnosticSource.pkgproj]
11:25:06 D:\j\workspace\AllConfigurat---c1564d53\Tools\Packaging.targets(424,5): error MSB4018: System.ArgumentException: An item with the same key has already been added. [D:\j\workspace\AllConfigurat---c1564d53\src\System.Diagnostics.DiagnosticSource\pkg\System.Diagnostics.DiagnosticSource.pkgproj]`

Could you please help with it?